### PR TITLE
[WIP] Setting up correclty strings to empty fragments

### PR DESCRIPTION
**Pull request by votaguz**
This pull request is intended to add default strings when fragments are empty correctly.

The idea is basically add to ApolloFragment a method called `getEmptyStringId()` that will return the default string id `R.string.empty_music`. 

Then that method will be overriden by subclasses of ApolloFragment such as `RecentFragment` or `LastAddedFragment` for example.

### CODE REVIEW

**Comment by votaguz** [comment]
@aldenml @gubatron noob question: Currently i'm having an issue with inheritance in this pull request. As you can see I created a method in `ApolloFragment` class called `getEmptyStringId()` and then did override it `RecentFragment` and `LastAddedFragment`.

At this point empty message is being showed correctly in RecentFragment but in `LastAddedFragment` the history is different. It keeps using the parent default empty string ( `R.string.empty_music` instead of `R.string.empty_last_added` ) 

Did i do something wrong ?

**Comment by aldenml** [comment]
I'm not a fan of this abstraction, looks to me like excessive use of OO polymorphism, but before going into that, are you sure the method removed is the right move? the method in the parent seems to do a lot of more

File: android/apollo/src/com/andrew/apollo/ui/fragments/RecentFragment.java [hunk 1]
```diff
@@ -88,4 +88,9 @@ public void onMetaChanged() {
             threadPool.execute(() -> restartLoader(true));
         }
     }
+
+    @Override
+    public int getEmptyStringId(){
+        return R.string.empty_recent;
```
Comment (gubatron) [inline android/apollo/src/com/andrew/apollo/ui/fragments/RecentFragment.java]: no need for this override, does the same as parent

**Comment by votaguz** [comment]
@aldenml, in a previous conversation @gubatron suggested to add an abstract method to be implemented for all the subclass of `ApolloFragment`. But I feel that it will be break the DIY principle. So, which one should be the path to take in this case?  My idea is to provide a better way to display that string.

**Comment by aldenml** [comment]
Think about it, what do you really want, provide an abstraction to customize a text, with all the disadvantages of providing more API to understand, or just tweak the text in the child `onLoadFinished`. That method can be simplified/reduced to just a couple of lines. Also remove the unnecessary `getString`s

**Comment by votaguz** [comment]
@aldenml @gubatron I come up with this solution to the problem. In first place i was modifying the wrong `LastAddedFragment`.
My new proposal is to declare a new field in parent class `ApolloFragment` and override that field in subclasses. If that field isn't overridden then use the default string.
Seems to be a more simpler way to do it, without declare a new method. 
I take this opportunity to ask: Why we have another `LastAddedFragment` under `profiles` directory ?   We really use this ? I was trying to do debugging in that class and debugger never stops in the method that i was testing. 🤔

**Comment by aldenml** [comment]
btw, take this back and forth only if you want to sharpen your criteria

**Comment by aldenml** [comment]
good point, is this a critical point performance wise?

**Comment by votaguz** [comment]
Well we have several fragments that should do the same when they don't have elements, thats my point of reference now, but i'm not sure how critical it can be. 

And now I have to ask, since we are using `findViewById` again, should we check for `null` return values ? I was checking the [docs](https://developer.android.com/reference/android/view/View.html#findViewById(int))  and says that: 

```Finds the first descendant view with the given ID, the view itself if the ID matches getId(), or null if the ID is invalid (< 0) or there is no matching view in the hierarchy.```

**Comment by aldenml** [comment]
yes, if going my route, the null check is necessary

**Comment by votaguz** [comment]
Ok, that's adds a new concern for me ( I'm talking about my n00b perspective ), we will have to repeat that four lines of code in every fragment that need to do that. 
```
        TextView empty = mRootView.findViewById(R.id.empty);
        if (empty != null) {
            empty.setText(R.string.empty_recent);
        }
```
And I feel that maybe we are breaking the DIY principle. Should worry about that at this point ?

**Comment by aldenml** [comment]
Ok, I'm willing to settle with this solution, I'm putting the comments in the code

**Comment by gubatron** [comment]
stupid note:
@votaguz -> DIY = Do it Yourself 😆 
I think you meant "DRY" ;)

File: android/apollo/src/com/andrew/apollo/ui/fragments/profile/ApolloFragment.java [hunk 2]
```diff
@@ -127,6 +127,8 @@
 
     protected ViewGroup mRootView;
 
+    protected Integer mDefaultFragmentEmptyString = null;
```
Comment (gubatron) [inline android/apollo/src/com/andrew/apollo/ui/fragments/profile/ApolloFragment.java]: why not an `int` and set it to `-1`?

File: android/apollo/src/com/andrew/apollo/ui/fragments/profile/ApolloFragment.java [hunk 3]
```diff
@@ -450,7 +452,8 @@ public void onLoadFinished(final Loader<List<I>> loader, final List<I> data) {
             // Set the empty text
             final TextView empty = mRootView.findViewById(R.id.empty);
             if (empty != null) {
-                empty.setText(getString(R.string.empty_music));
+                empty.setText((mDefaultFragmentEmptyString != null) ?
```
Comment (gubatron) [inline android/apollo/src/com/andrew/apollo/ui/fragments/profile/ApolloFragment.java]: then here `(mDefaultFragmentEmptyString != -1)`

**Comment by gubatron** [comment]
LGTM

**Comment by gubatron** [comment]
will add this to `487`

**Pull request merged**